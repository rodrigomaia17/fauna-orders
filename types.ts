export interface Order {
  _id: string;
  status: string;
  shipAddress: Address;
  line: Line[];
  customer: Customer;
  totalPrice: number;
}

export interface Customer {
  name: string;
}

export interface Product {
  name: string;
  description: string;
}

export interface Line {
  product: Product;
  quantity: number;
  price: number;
  total: number;
}

export interface Address {
  zipCode: string;
  street: string;
  city: string;
  state: string;
}
