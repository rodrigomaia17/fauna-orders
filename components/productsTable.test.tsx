import React from 'react';
import { render } from '@testing-library/react';
import ProductsTable from './productsTable';
import { Line } from '../types';
import '@testing-library/jest-dom/extend-expect';

const lines: Line[] = [
  {
    product: {
      name: 'product Name',
      description: 'description'
    },
    quantity: 1,
    price: 2,
    total: 2
  }
];

test('print the table passed as param', () => {
  const { getByText, container } = render(<ProductsTable lines={lines} />);

  expect(getByText('product Name')).toBeVisible();
  expect(container.firstChild).toMatchSnapshot();
});
