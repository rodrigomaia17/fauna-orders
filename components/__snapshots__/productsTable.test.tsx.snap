// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print the table passed as param 1`] = `
<div
  class="container"
>
  <p>
    Products 
  </p>
  <div
    class="table-container"
  >
    <table
      class="table is-narrow is-hoverable is-striped"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Description
          </th>
          <th>
            Quantity
          </th>
          <th>
            Price
          </th>
          <th>
            Subtotal
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="has-text-weight-light"
        >
          <td>
            product Name
          </td>
          <td>
            description
          </td>
          <td>
            1
          </td>
          <td>
            2.00
          </td>
          <td>
            2.00
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
